sdks/python/apache_beam/testing/util.py

```diff
@@ -46,6 +56,26 @@ class BeamAssertException(Exception):
   pass
 
 
+# Used for reifying timestamps and windows for assert_that matchers.
```

Comment:
I think we can get around the need for introducing this new class by pairing values with None internally on reification (or at least making it an internal implementation detail).

Reply:
The idea was to use this class for user-implemented matchers (inheriting from WindowedValueMatcher). Am I wrong to assume that there are matchers defined outside this repo?

Reply:
Ideally one should be able to use any callable, e.g. hamcrest matchers, rather than have to implement a windowed variant of each. I misspoke about having a windowed_equals_to, we should have a `assert_that_windowed(pcoll, equal_to([WindowedValue(...), ...]))`, or `assert_that(pcoll, equal_to([WindowedValue(...), ...], reify_windows=True)`.

Reply:
done.